Allow feedin capping for DSO

Bachibouzouk opened the pull request:
Prior to merge this PR #939  should be merged

Fix #941 

**Changes proposed in this pull request**:
- Add functions to format name of automatically created assets in DSO setting for feedin cap and peak demand pricing
- Enable feedin cap by adding a transformer and an extra bus between existing bus and DSO feedin
- Add a special character to name of busses created automatically

The following steps were realized, as well (if applies):
- [x] Use in-line comments to explain your code
- [x] Write docstrings to your code ([example docstring](https://multi-vector-simulator.readthedocs.io/en/latest/Developing.html#format-of-docstrings))
- [ ] For new functionalities: Explain in readthedocs
- [ ] Write test(s) for your new patch of code (pytests, assertion debug messages)
- [ ] Update the CHANGELOG.md
- [x] Apply black (`black . --exclude docs/`)
- [x] Check if benchmark tests pass locally (`EXECUTE_TESTS_ON=master pytest`)


<sub>*For more information on how to contribute check the [CONTRIBUTING.md](https://github.com/rl-institut/multi-vector-simulator/blob/dev/CONTRIBUTING.md).*<sub>

Bachibouzouk commented:
## Uncapped system

![energy_system_graph](https://user-images.githubusercontent.com/4399407/162328923-865d860b-7771-43e6-8290-9e86b83236d1.png)



## Capped system

![energy_system_graph](https://user-images.githubusercontent.com/4399407/162328854-49c2bc9d-c4fc-4261-bb59-b18542805323.png)

Bachibouzouk commented:
@ciaradunks, @smartie2076  what do you think?

I tested it and could cap the feedin successfully. Now I will modify RTD, add a test and maybe there are post-processing calculations which need to change? Like the power in/out of the feedin_cap bus is not interesting to look at as the information is already there from flow from normal electricity bus to transformer station.

smartie2076 commented:
I am not 100% sure why you need this instead of the capping through an additional transformer station between the local energy bus and the energy provider bus. From the idea, yes, this should work.

If you also introduced a consumption capping, you would symetrical features for consumption and feed-in. In the long run, this could also help to implement the issue #940, I think.

> Now I will modify RTD, add a test and maybe there are post-processing calculations which need to change? Like the power in/out of the feedin_cap bus is not interesting to look at as the information is already there from flow from normal electricity bus to transformer station.

This is something I also dont know from the top of my head anymore, so I am just having a good guess here.
I think it there are no costs connected with the feedin CAP, you might not have to change stuff? Maybe you should write something for future devs that if the feedin cap must be optimized, and in that scope the asset gets costs, then the economic equations should be checked again.
Regarding the technical parameters: It should not change how flows of the energy provider are accessed...

Bachibouzouk commented:
> I am not 100% sure why you need this instead of the capping through an additional transformer station between the local energy bus and the energy provider bus. From the idea, yes, this should work.

Isn't it what I implemented? Adding a transformer to cap the feedin. The reason I implemented this is because in open_plan the user should not need to know that one can cap by adding an "oemof transformer", we removed the ability to use oemof transformers directly in the GUI because the users had confused them with electrical transformers. The user only wants to set a capping at the DSO level and does not care what happens under the hood of MVS, this is why I implemented this.
